doc/bucket-replication.md

```diff
@@ -0,0 +1,97 @@
+    - (Optional, log-based optimization) `sync_versions` - can be set to a boolean value to indicate whether object versions should be replicated (i.e. if the source bucket has versioning enabled, the target bucket will also have versioning enabled, and all object versions will be synced)
+
+In addition, when the bucketclass is backed by namespacestores, each policy can be set to optimize replication by utilizing logs (configured and supplied by the user, currently only supports AWS S3 and Azure Blob):
+  - (Optional, only supported on namespace buckets) `log_replication_info` - an object that contains data related to log-based replication optimization -
+    - (Necessary on Azure) `endpoint_type` - this field can be set to an appropriate endpoint type (currently, only AZURE is supported)
+    - (Necessary on AWS) `{"logs_location": {"logs_bucket": <>}}` - this field should be set to the location of the AWS S3 server access logs
+
+## Examples
+Note that the example poicies below can also be saved as files and passed to the NooBaa CLI. In that case, it's necessary to omit the outer single quotes.
+### AWS replication policy:
+
+`'{"rules":[{"rule_id":"aws-rule-1", "destination_bucket":"first.bucket", "filter": {"prefix": "a."}}]}'`
+
+### AWS replication policy with log optimization:
+
+`'{"rules":[{"rule_id":"aws-rule-1", "destination_bucket":"first.bucket", "filter": {"prefix": "a."}}], "log_replication_info": {"logs_location": {"logs_bucket": "logsarehere"}}}'`
+
+### Azure replication policy with log optimization, deletion and version sync:
+
+`'{"rules":[{"rule_id":"azure-rule-1", "sync_deletions": true, "sync_versions": false, "destination_bucket":"first.bucket"}], "log_replication_info": {"endpoint_type": "AZURE"}}'`
+
+### Namespace bucketclass creation with replication to first.bucket:
+
+With the NooBaa CLI -
+
+```shell
+noobaa -n app-namespace bucketclass create namespace-bucketclass single bc --resource azure-blob-ns --replication-policy=/path/to/json-file.json
+```
+/path/to/json-file.json is the path to a JSON file which defines the replication policy, e.g. -
+```json
+{"rules":[{ "rule_id": "rule-1", "destination_bucket": "first.bucket", "filter": {"prefix": "d"}} ]}
+```
+
+With a YAML file:
+```yaml
+apiVersion: noobaa.io/v1alpha1
+kind: BucketClass
+metadata:
+  name: bc
+  namespace: app-namespace
+spec:
+  namespacePolicy:
+    type: Single
+    single: 
+      resource: azure-blob-ns
+  replicationPolicy: '[{ "rule_id": "rule-1", "destination_bucket": "first.bucket", "filter": {"prefix": "ba"}}]'
+```
+
+### OBC with specific Replication Policy
+
+Applications that require a bucket to have a specific replication policy can create an OBC and add to the claim 
+the `spec.additionalConfig.replication-policy` property.
+
+/path/to/json-file.json is the path to a JSON file which defines the replication policy
+
+Example:
```

Comment:
Here as well (if you can add the JSON file, and add a sentence if there are 2 optional ways or that you are showing the result in the YAML file...).